modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/display/context/DataSetDisplayContext.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ * <p>
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * <p>
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.sample.web.internal.display.context;
+
+import com.liferay.frontend.taglib.clay.data.set.servlet.taglib.util.ClayDataSetActionDropdownItem;
+import com.liferay.frontend.taglib.clay.sample.web.internal.display.context.util.DataSetDisplayRequestHelper;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.CreationMenu;
+import com.liferay.petra.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.portlet.PortletURLUtil;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.portlet.PortletException;
+import javax.portlet.PortletURL;
+
+
+
+/**
+ * @author Javier Gamarra
+ */
+public class DataSetDisplayContext {
+
+	private final DataSetDisplayRequestHelper _dataSetDisplayRequestHelper;
+
+	public String getAPIURL() {
+		return "/o/object-admin/v1.0/object-definitions";
+	}
+
+
+	public DataSetDisplayContext(
+		HttpServletRequest httpServletRequest) {
+
+		_dataSetDisplayRequestHelper = new DataSetDisplayRequestHelper(httpServletRequest);
+	}
+
+	public List<ClayDataSetActionDropdownItem>
+		getClayDataSetActionDropdownItems()
+		throws Exception {
+
+		return new ArrayList<>();
+	}
+
+	public CreationMenu getCreationMenu() throws Exception {
+		CreationMenu creationMenu = new CreationMenu();
+
+		return creationMenu;
+	}
+
+	public PortletURL getPortletURL() throws PortletException {
+		return PortletURLUtil.clone(
+			PortletURLUtil.getCurrent(
+				_dataSetDisplayRequestHelper.getLiferayPortletRequest(),
+				_dataSetDisplayRequestHelper.getLiferayPortletResponse()),
+			_dataSetDisplayRequestHelper.getLiferayPortletResponse());
```

Comment:
This looks very awkward. A `*DisplayContext` can have `liferayPortletRequest` and `liferayPortletResponse` as constructor arguments, like for example in [ClaySampleManagementToolbarsDisplayContext](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/display/context/ClaySampleManagementToolbarsDisplayContext.java#L37-L44). 

Do we need `getPortletURL()`? It looks to me like we are not using it at all.

Do we need `DataSetDisplayRequestHelper`? It looks to me like this is the only place it is used.

Reply:
We just find the pattern used by commerce and translate it to the module. I'll dig a little bit if some things can be simplified